src/rule_evaluator.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { OperatorType, Rule, RuleType } from './rule';
+import { matchesAnyRule } from './rule_evaluator';
+
+describe('matchesAnyRule', () => {
+  const emptyConditionsRule: Rule = {
+    type: RuleType.AND,
+    conditions: [],
+  };
+  const numericRuleAnd: Rule = {
+    type: RuleType.AND,
+    conditions: [
+      {
+        operator: OperatorType.GTE,
+        attribute: 'totalSales',
+        value: 10,
+      },
+      {
+        operator: OperatorType.LTE,
+        attribute: 'totalSales',
+        value: 100,
+      },
+    ],
+  };
+  const textRuleOr: Rule = {
+    type: RuleType.AND,
+    conditions: [
```

Comment:
I only see one condition here, but the type is `AND`. The `type` applies to the `conditions` within this object (as opposed to rules outside of this object), is that right?

Reply:
Right, applies to all conditions.